### PR TITLE
Forced bits infrastructure improvements

### PR DESCRIPTION
**Pull request by manasij7479**

### CODE REVIEW

**Comment by manasij7479** [comment]
Just noticed that a test is failing. Don't merge.

**Comment by manasij7479** [comment]
Test seems to pass locally, will figure out what's going wrong here.

File: include/souper/Infer/AbstractInterpreter.h [hunk 1]
```diff
@@ -201,12 +201,20 @@ namespace souper {
...
     }
-    struct Value {
-      llvm::APInt Concrete;
-      // Add Range and Knownbits, maybe
+    class Value {
+    public:
+      Value(llvm::APInt Val_) : Val{Val_} {}
       bool conflict(Value &Other) {
-        return Concrete != Other.Concrete;
+        if (hasConcrete() && Concrete() != Other.Concrete()) {
```
Comment (regehr) [inline include/souper/Infer/AbstractInterpreter.h:208]: don't write code of the form

if (x) return true; else return false;

go back to the previous syntax, and just change the condition

File: include/souper/Infer/AbstractInterpreter.h [hunk 2]
```diff
@@ -201,12 +201,20 @@ namespace souper {
...
+          return true;
+        }
+        return false;
       }
+      bool hasConcrete() {return Val.hasValue();}
```
Comment (regehr) [inline include/souper/Infer/AbstractInterpreter.h:213]: put a space after the { and before the }

**Comment by regehr** [comment]
LGTM thanks

**Pull request merged**